python/lsst/daf/butler/core/datastore.py

```diff
@@ -773,6 +773,13 @@ def transfer_from(
             this method with details of all artifacts tested. Can be `None`
             if the caller is not interested.
 
+        Returns
+        -------
+        accepted : `set` [`DatasetRef`]
+            The datasets that were transferred.
+        rejected : `set` [`DatasetRef`]
+            The datasets that were rejected due to a constraints violation.
```

Comment:
Good question. It seems like I'm currently rejecting the target ref and not the source ref but since we are using UUID they are the same thing. The real answer is that I need to change things a bit to now understand that there is no difference between source and target ref.

1. Remove the id_gen_map parameter from Butler.transfer_from (they are always UUID)
2. Remove the local_refs parameter from Datastore.transfer_from.

This will make things a lot simpler. I will see if I can quickly remove the relevant code -- if it's just removing code I won't send it for a new review.

Reply:
Cleaning up to assume UUID is quite involved. Will put on a new ticket.